flooddrake/timestepper.py

```diff
@@ -0,0 +1,309 @@
+        NegFlux = Fluxes(self.N('-'), self.V)
+        neg_flux = NegFlux.Interior_Flux(w_minus, w_plus)
+        boundary_flux = PosFlux.Boundary_Flux(w)
+
+        # Define source term
+        if self.mesh.geometric_dimension() == 1:
+            source = as_vector((0, gravity * h * self.b_.dx(0)))
+        if self.mesh.geometric_dimension() == 2:
+            source = as_vector(
+                (0, gravity * h * self.b_.dx(0), gravity * h * self.b_.dx(1)))
+
+        ####################### SOLVER ######################
+
+        if self.mesh.geometric_dimension() == 2:
+            L = dot(self.v,
+                    ((w_ - w) / self.dt)) * dx - dot(self.v.dx(0),
+                                                     F1) * dx - dot(self.v.dx(1),
+                                                                    F2) * dx + (dot(self.v('-'),
+                                                                                    neg_flux) + dot(self.v('+'),
+                                                                                                    pos_flux)) * dS + dot(self.v,
+                                                                                                                          boundary_flux) * ds + (dot(source,
+                                                                                                                                                     self.v)) * dx - (dot(self.v('-'),
+                                                                                                                                                                          delta_minus) + dot(self.v('+'),
+                                                                                                                                                                                             delta_plus)) * dS
+        if self.mesh.geometric_dimension() == 1:
+            L = dot(self.v,
+                    ((w_ - w) / self.dt)) * dx - dot(self.v.dx(0),
+                                                     F) * dx + (dot(self.v('-'),
+                                                                    neg_flux) + dot(self.v('+'),
+                                                                                    pos_flux)) * dS + dot(self.v,
+                                                                                                          boundary_flux) * ds + (dot(source,
+                                                                                                                                     self.v)) * dx - (dot(self.v('-'),
+                                                                                                                                                          delta_minus) + dot(self.v('+'),
+                                                                                                                                                                             delta_plus)) * dS
+
```

Comment:
autopep8 has gone a bit made with the line breaks in various places.  You may wish to go through and make this indentation nicer, for example, for this guy, I would probably have:

```python
L = ((dot(source, self.v) -
      dot(self.v, ((w_ - w) / self.dt)) - 
      dot(self.v.dx(0), F))*dx +
     dot(self.v, boundary_flux)*ds +
     (dot(self.v('-'), neg_flux) + dot(self.v('+'), pos_flux) +
      dot(self.v('-'), delta_minus) + dot(self.v('+'), delta_plus))*dS)
```